### PR TITLE
Prefer `Typographic Family` over `Family`

### LINKED ISSUE #35: Prefer Typographic Family to Family Name when available.

### PR DESCRIPTION
**Pull request by CryZe**
The `Typographic Family` (ID 16) is sometimes also called the `Preferred Family` and was created to handle various limitations of the `Family` (ID 1). It is recommended that applications support both of them. It is also required to handle variable fonts properly.

Resolves #35

### CODE REVIEW

**Comment by CryZe** [comment]
I also tested it with `cosmic-text` and everything works nicely now.

**Comment by RazrFalcon** [comment]
Yep, this should do. Thanks!

**Comment by CryZe** [comment]
Let me quickly do another quick fix.

**Comment by RazrFalcon** [comment]
Ok.

File: src/lib.rs [hunk 1]
```diff
@@ -935,10 +923,10 @@ fn parse_names(raw_face: &ttf_parser::RawFace) -> Option<(Vec<(String, Language)
         if let Some(index) = families
             .iter()
             .position(|f| f.1 == Language::English_UnitedStates)
-            .filter(|index| *index > 0)
         {
-            let v = families.remove(index);
-            families.insert(0, v)
+            if index != 0 {
```
Comment (CryZe) [inline src/lib.rs:927]: I quickly moved the index check down, as otherwise it might skip over the first English name and still swap the second one in.

Comment (RazrFalcon) [reply src/lib.rs:927]: Are you sure? The idea is that if an English name isn't the first one (which happens), then we have to move to the first place.

Comment (CryZe) [reply src/lib.rs:927]: Oh wait, I thought that was Iterator::filter, not Option::filter. My if basically does the same as your previous Option::filter call then, so there's no change in behavior. I can revert that part if you want. (Though arguably the filter call is a little confusing as it's part of an iterator chain and not the actual Iterator::filter method)

Comment (RazrFalcon) [reply src/lib.rs:927]: Sure, let's leave as `if` condition for readability.

**Pull request merged**